cmake/Extensions.cmake

```diff
@@ -22,10 +22,31 @@ define_property(GLOBAL PROPERTY EXTENSION-OPTIONS
 
 set_property(GLOBAL PROPERTY EXTENSION-OPTIONS "")
 
+set(extension-build-info-file "${CMAKE_CURRENT_BINARY_DIR}/ExtensionBuildInfo.cpp")
+file(GENERATE OUTPUT ${extension-build-info-file}
+    CONTENT "\
+    #include \"utils/Export.h\"\n\
+    #ifdef BUILD_ID_VARIABLE_NAME\n\
+    EXTENSIONAPI extern const char* const BUILD_ID_VARIABLE_NAME = \"__EXTENSION_BUILD_IDENTIFIER_BEGIN__${BUILD_IDENTIFIER}__EXTENSION_BUILD_IDENTIFIER_END__\";\n\
```

Comment:
I think using `extern` here is incorrect, since it only declares, but doesn't define a variable. That is, unless the variable is defined somewhere. How it ends up in the binary is not clear to me, it shouldn't be there.

Reply:
could you expand on this? should this be split into a separate declaration with `extern` and a definition? I found that without `extern` this const variable would have internal linkage and stripped from the shared library

Reply:
Normally `extern` is used do declare a global variable in a header that is then defined in some implementation file without violation ODR. Otherwise a global variable declaration would also be a definition. I don't understand why having an initializer here is even legal c++, since it should go to the separate definition, not the declaration.

According to [this](https://stackoverflow.com/a/57900212) stackoverflow answer, having `extern` here is the same as not having it, because the initializer makes a definition and all globals have external linkage by default.

What do you think about having a global function instead that returns a `const char*` to the magic string? I think that shouldn't be stripped from the binary even with LTO.

Reply:
only non-const variables are global by default, const ones have internal linkage unless declared extern

Reply:
Okay, I missed that. Now it all makes sense, nevermind the comment.